table_processing/GeneratedTable.py

```diff
@@ -28,25 +37,66 @@ def generate_df(self,rows):
         self.df = myDB.gen_dataframe(
         rows,fields=['name','city','phone',
         'license_plate','email'],
-        real_email=False,phone_simple=True
-        )
-          
+        real_email=False,phone_simple=True)
 
+        if self.multi_row == True:
+            for row in self.df.index:
+                if row % 3 ==0:
+                    self.df.loc[row,self.df.columns[0]] = ''
+        
     def to_pdf(self):
-        doc = pl.Document(geometry_options=geometry_options)
+        doc = pl.Document(geometry_options=self.geometry_options)  
 
 
-        if self.row_lines == True:
-            with doc.create(pl.Section('Table')):
-                with doc.create(pl.LongTable(self.table_spec, row_height=self.row_height)) as table:
-                    #table.add_hline()
+        if self.row_lines == True and self.multi_row == False:
+            
+            with doc.create(pl.Center()) as centered:
+                with centered.create(pl.LongTable(self.table_spec, row_height=self.row_height)) as table:
                     table.add_row(list(self.df.columns))
                     table.add_hline()
                     for row in self.df.index:
-                        print(row)
                         table.add_row(list(self.df.loc[row,:]))
                         table.add_hline()
-                    #table.add_hline()
+   
+
+                self.filename = sd.uuid()
+
+                self.path = './generated_tables/' + self.filename 
+
+                doc.generate_pdf(self.path, compiler='pdflatex')
+                self.path += '.pdf'
+
```

Comment:
I am concerned that this gets to be a very deep level of nested looping and if statements. When this happens it tends to point to a need to refactor, perhaps breaking up the logic into multiple functions or other.